src/main/java/com/cdancy/jenkins/rest/parsers/LocationToQueueId.java

```diff
@@ -17,33 +17,47 @@
 
 package com.cdancy.jenkins.rest.parsers;
 
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 import javax.inject.Singleton;
 
 import org.jclouds.http.HttpResponse;
 
+import com.google.common.base.Charsets;
 import com.google.common.base.Function;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+import com.google.common.io.CharStreams;
+
+import com.cdancy.jenkins.rest.domain.common.Error;
+import com.cdancy.jenkins.rest.domain.common.IntegerResponse;
 
 /**
  * Created by dancc on 3/11/16.
  */
 @Singleton
-public class LocationToQueueId implements Function<HttpResponse, Integer> {
+public class LocationToQueueId implements Function<HttpResponse, IntegerResponse> {
 
    private static final Pattern pattern = Pattern.compile("^.*/queue/item/(\\d+)/$");
 
-   public Integer apply(HttpResponse response) {
+   public IntegerResponse apply(HttpResponse response) {
 
       String url = response.getFirstHeaderOrNull("Location");
+      List<Error> errors = Lists.newArrayList();
       if (url != null) {
          Matcher matcher = pattern.matcher(url);
          if (matcher.find() && matcher.groupCount() == 1) {
-            return Integer.valueOf(matcher.group(1));
+            return IntegerResponse.create(Integer.valueOf(matcher.group(1)), errors);
          }
       }
-
-      return 0;
+      errors.add(Error.create("No context",
+         "No queue item Location header could be found despite getting a valid HTTP response.",
+         "None"));
```

Comment:
Lets take the `Error.create` call outside and do something like:
```
final Error error = Error.create(null,
    "No queue item Location header could be found despite getting a valid HTTP response.", 
    NumberFormatException.class.getCanonicalName());
List<Error> errors = Lists.newArrayList(error);
return IntegerResponse.create(null, errors);
```

I'd prefer to delay the creation of the list until the last possible minute as this is an exceptional case. In the code further above lets just pass a null like so:
```
        return IntegerResponse.create(Integer.valueOf(matcher.group(1)), null);

```